### PR TITLE
[chore] app_container_config: simplify interface further

### PR DESCRIPTION
**Pull request by jevonyeoh**
## Description
This PR updates the app container further, to simplify the interface in registering services to the config, to just a single method of `register_class`. We achieve this by pre-populating the instance_map with the `user`, `persistent` and `catalog_id` so that services registered via `register_service`, can now also just look up the user, persistent and catalog_id during construction like any other dependency.

We also make a small change to how names are parsed. Previously, a class with multiple upper case letters in a row will be parsed as such `SCDTable` -> `s_c_d_table`. This caused some errors when we tried to look up the dependencies, as we normally name our variables in the format `scd_table` instead. This change should alleviate the need for name_override's in this scenario.

## Type of Change

Label this pull request to place it under the correct category in Release Notes:

|        **Pull Request Label**         | **Category in Release Notes** |
|:-------------------------------------:|:-----------------------------:|
|       `enhancement`, `feature`        |          🚀 Features          |
| `bug`, `refactoring`, `bugfix`, `fix` |    🔧 Fixes & Refactoring     |
|       `build`, `ci`, `testing`        |    📦 Build System & CI/CD    |
|              `breaking`               |      💥 Breaking Changes      |
|            `documentation`            |       📝 Documentation        |
|            `dependencies`             |    ⬆️ Dependencies updates    |

## Checklist

- [ ] I have read the [`CODE_OF_CONDUCT.md`](https://github.com/featurebyte/featurebyte/blob/main/CODE_OF_CONDUCT.md) and [`CONTRIBUTING.md`](https://github.com/featurebyte/featurebyte/blob/main/CONTRIBUTING.md) guides.
- [ ] I have written tests for the changes made.
- [ ] I have written docstrings in [NumpyDoc format](https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard)
- [ ] I have labeled my Pull Request correctly

### CODE REVIEW

File: featurebyte/routes/registry.py [hunk 1]
```diff
@@ -102,131 +103,145 @@
...
+    This is a special placeholder class that is used to inject the container config.
+    """
+
+
+# This looks a little funny right now, but every entry in the instance map must currently be found in the
+# app_container_config, as some validation checks depend on it. For this particular case, we inject the
+# catalog_id directly into the `instance_map` in the LazyAppContainer constructor. As such, we need an item in the
+# app_container_config that is called `catalog_id`. This class of CatalogId will get parsed into `catalog_id`, and
+# as such, works as a placeholder.
+app_container_config.register_class(CatalogId, force_no_deps=True)
```
Comment (jevonyeoh) [inline featurebyte/routes/registry.py:244]: i'll try to clean this up in a follow-up, but works ok for now, and isn't too common.

File: tests/unit/routes/test_lazy_app_container.py [hunk 2]
```diff
@@ -112,18 +112,6 @@ def test_lazy_initialization(test_app_config, app_container_constructor_params):
     assert "test_controller" in instance_map
 
 
-def test_construction__empty_app_config_has_two_instances(app_container_constructor_params):
```
Comment (jevonyeoh) [inline tests/unit/routes/test_lazy_app_container.py:115]: this test wasn't really doing much, so removing it.

File: featurebyte/routes/app_container_config.py [hunk 3]
```diff
@@ -166,12 +125,21 @@ def register_class(
...
-            dependencies
+        dependency_override: Optional[Dict[str, str]]
+            We will normally look up dependencies by the name of the variable specified in the constructor of the
+            class that we're registering. You can provide an override if you want to explicitly specify a class
+            that we should inject instead. This is common when trying to initialize a class that inherits a constructor
+            from a parent class, and the name in the constructor is something generic. For example, within our repo,
+            simple controllers will typically inherit a constructor that has a parameter called "service". This will
+            typically be the service that corresponds to the controller. However, since service is just a generic name,
+            we can provide an override from `service` -> `controllers_service` to tell the dependency injector to
+            look for `controllers_service` instead when trying to initialize this controller.
```
Comment (gee-senbong) [inline featurebyte/routes/app_container_config.py:136]: If i understood this correctly, this is used to handle the case when the first service of the controller is not `service` (for example, in `ContextController`, the first service is `context_service`, right?)

**Review by gee-senbong — APPROVED** [review]
LGTM.

File: featurebyte/routes/app_container_config.py (hunk 3 above)
Comment (jevonyeoh) [reply featurebyte/routes/app_container_config.py:136]: yeap!

**Pull request merged**